server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainMappingsRoutesTest.java

```diff
@@ -0,0 +1,387 @@
+
+            given()
+                .body("to.com")
+            .when()
+                .delete("from.com")
+            .then()
+                .body(isEmptyString());
+
+            assertThat(recipientRewriteTable.getAllMappings()).isEmpty();
+        }
+
+        @Test
+        void getSpecificDomainMappingShouldRespondWithNotFoundWhenHasNoAliases() {
+            String domain = "from.com";
+
+            when()
+                .get(domain)
+            .then()
+                .contentType(ContentType.JSON)
+                .statusCode(HttpStatus.NOT_FOUND_404)
+                .body("type", is(ErrorResponder.ErrorType.NOT_FOUND.getType()))
+                .body("statusCode", is(HttpStatus.NOT_FOUND_404))
+                .body("message", is("Cannot find mappings for " + domain));
+        }
+
+        @Test
+        void getSpecificDomainMappingShouldRespondWithNotFoundWhenHasEmptyAliases() throws RecipientRewriteTableException {
+            String domain = "from.com";
+
+            Map<MappingSource, Mappings> allMappings = ImmutableMap.of(
+                    MappingSource.fromDomain(Domain.of(domain)),
+                    MappingsImpl.empty()
+            );
+
+            when(recipientRewriteTable.getAllMappings()).thenReturn(allMappings);
+
+            when()
+                .get(domain)
+            .then()
+                .contentType(ContentType.JSON)
+                .statusCode(HttpStatus.NOT_FOUND_404)
+                .body("type", is(ErrorResponder.ErrorType.NOT_FOUND.getType()))
+                .body("statusCode", is(HttpStatus.NOT_FOUND_404))
+                .body("message", is("Cannot find mappings for " + domain));
+        }
+
+        @Test
+        void getSpecificDomainMappingNotDomainsMappingsShouldBeFilteredOut() throws RecipientRewriteTableException {
+            String domain = "from.com";
+            String aliasDomain = "to.com";
+            final MappingSource mappingSource = MappingSource.fromDomain(Domain.of(domain));
+
+            recipientRewriteTable.addRegexMapping(mappingSource, "(.*)@localhost");
+            recipientRewriteTable.addGroupMapping(mappingSource, "user@domain.com");
+            recipientRewriteTable.addForwardMapping(mappingSource, "user@domain.com");
+            recipientRewriteTable.addErrorMapping(mappingSource, "disabled");
+            recipientRewriteTable.addAliasDomainMapping(mappingSource, Domain.of(aliasDomain));
+
+            List<String> body =
+                    when()
+                        .get(domain)
+                    .then()
+                        .contentType(ContentType.JSON)
+                        .statusCode(HttpStatus.OK_200)
+                    .extract()
+                        .jsonPath()
+                        .getList(".");
+
+            assertThat(body).containsOnly(aliasDomain);
+        }
+
+        @Test
+        void getSpecificDomainMappingShouldResponseWithOK() throws RecipientRewriteTableException {
```

Comment:
I'd say here you test more than 'OK' response. `getSpecificDomainMappingShouldReturnDomainMappings`